lib/pbench/test/unit/server/test_upload.py

```diff
@@ -493,3 +512,68 @@ def setvalue(dataset: Dataset, key: str, value: Any):
         assert audit[1].user_name == "drb"
         assert audit[1].reason == AuditReason.INTERNAL
         assert audit[1].attributes == {"message": "INTERNAL ERROR"}
+
+    @pytest.mark.freeze_time("1970-01-01")
+    def test_upload_archive(
+        self, client, pbench_drb_token, server_config, setup_ctrl, tarball
+    ):
+        """Test a successful archiveonly dataset upload."""
+        datafile, _, md5 = tarball
+        with datafile.open("rb") as data_fp:
+            response = client.put(
+                self.gen_uri(server_config, datafile.name),
+                data=data_fp,
+                headers=self.gen_headers(pbench_drb_token, md5),
+                query_string={"metadata": "server.archiveonly:t,server.origin:test"},
+            )
+
+        assert response.status_code == HTTPStatus.CREATED, repr(response.data)
+        name = Dataset.stem(datafile)
+
+        dataset = Dataset.query(resource_id=md5)
+        assert dataset is not None
+        assert dataset.resource_id == md5
+        assert dataset.name == name
+        assert dataset.uploaded.isoformat() == "1970-01-01T00:00:00+00:00"
+        # assert Metadata.getvalue(dataset, Metadata.ARCHIVEONLY) is True
+        assert Metadata.getvalue(dataset, Metadata.ORIGIN) == "test"
+        assert Metadata.getvalue(dataset, Metadata.DELETION) == "1972-01-02"
+        assert Metadata.getvalue(dataset, "dataset.operations") == {
+            "BACKUP": {"state": "READY", "message": None},
+            "UPLOAD": {"state": "OK", "message": None},
+        }
+        assert self.cachemanager_created
+        assert dataset.name in self.cachemanager_created
+
+        audit = Audit.query()
+        assert len(audit) == 2
+        assert audit[0].id == 1
+        assert audit[0].root_id is None
+        assert audit[0].operation == OperationCode.CREATE
+        assert audit[0].status == AuditStatus.BEGIN
+        assert audit[0].name == "upload"
+        assert audit[0].object_type == AuditType.DATASET
+        assert audit[0].object_id == md5
+        assert audit[0].object_name == name
+        assert audit[0].user_id == "3"
```

Comment:
`DRB_USER_ID` instead of 3? ditto on line 573

Reply:
This just means you blew it when you made the earlier test changes: this was cloned from another test case. Now you're making *me* clean up your mess! 😆

Reply:
Are you cleaning up the source of the clone too?  😬

Reply:
Up to Dave but I am gonna resolve this as this is fixed :smile:

Reply:
I think he _did_ fix both.